autoblocks/_impl/tracer.py

```diff
@@ -244,6 +231,66 @@ async def _send_event_unsafe(
         )
         req.raise_for_status()
 
+    def _send_test_event_unsafe(
+        self,
+        # Require all arguments to be specified via key=value
+        *,
+        message: str,
+        timestamp: str,
+        properties: Dict[Any, Any],
+        trace_id: Optional[str],
+    ) -> None:
+        """Sends an event to the Autoblocks CLI in a testing environment."""
+        if not self._cli_server_address:
+            log.error("Failed to send test event to Autoblocks. CLI server address not set.")
+            return
+
+        if not current_external_test_id.get():
+            log.error("Failed to send test event to Autoblocks. No test ID set.")
+            return
+
+        if not current_test_case_hash.get():
+            log.error("Failed to send test event to Autoblocks. No test case hash set.")
+            return
+
+        traced_event = TracerEvent(
+            message=message,
+            trace_id=trace_id,
+            timestamp=timestamp,
+            properties=properties,
+        )
+        req = global_state.sync_http_client().post(
+            url=f"{self._cli_server_address}/events",
+            json=dict(
+                testExternalId=current_external_test_id.get(),
+                testCaseHash=current_test_case_hash.get(),
+                event=traced_event.to_json(),
+            ),
+            timeout=self._timeout_seconds,
+        )
+        req.raise_for_status()
+
+    def merge_properties(
```

Comment:
can you add a leading underscore since this is an internal function

Reply:
ya i already did if you refresh 😄